install/kubernetes/cilium/values.yaml.tmpl

```diff
@@ -2560,8 +2560,19 @@ clustermesh:
       digest: ${CLUSTERMESH_APISERVER_DIGEST}
       useDigest: ${USE_DIGESTS}
       pullPolicy: "${PULL_POLICY}"
+
     # -- TCP port for the clustermesh-apiserver health API.
     healthPort: 9880
+    readinessProbe:
+      # -- failure threshold of the readiness probe
+      failureThreshold: 3
+      # -- interval between checks of the readiness probe
+      periodSeconds: 30
+      # -- duration to wait before the first probe
+      initialDelaySeconds: 5
+      # -- timeout for the readiness probe
+      timeoutSeconds: 5
```

Comment:
Any reason for not using kubernetes default values here? (Same for kvstoremesh). I'd also configure at least `contrib/testing/kind-clustermesh{1,2}.yaml` with more aggressive values (e.g. `1 second period`).

Reply:
The default for `periodsSeconds` is 10, which seems too aggressive for a check that we really only need to pass once but will continue to run during the lifetime of the Pod. It would be more convenient if there was something similar to a startupProbe but for readiness. Default for initial delay is 0, which would likely cause the pod to take at least 30 seconds to become ready in most cases.

The other 2 are the same as the default setting, but it doesn't seem like a bad thing to have those values configurable with the rest.

Reply:
> The default for periodsSeconds is 10, which seems too aggressive for a check that we really only need to pass once but will continue to run during the lifetime of the Pod.

I'd personally argue that this is the case with quite a few readiness probes (and regardless of whether it is expected to change or not, the amount of requests is the same), and a local HTTP request every 10 seconds should be more than fine, especially given that the handler is practically a no-op in terms of overhead. But that's mostly a matter of opinions I guess.

> It would be more convenient if there was something similar to a startupProbe but for readiness.

Totally agree. It should be pretty easy to configure a `startupProbe` for kvstoremesh, given the global circuit breaker timeout, but it is way more trickier for the clustermesh-apiserver. And overall the risks associated with the startupProbe terminating the containers overweight the advantages.

> Default for initial delay is 0, which would likely cause the pod to take at least 30 seconds to become ready in most cases.

The initial delay doesn't appear to work as I (and based on the above also you) expected. Probing is performed according to the usual schedule, but [the result is ignored if it falls during the initial delay window](https://github.com/kubernetes/kubernetes/blob/4b8e819355d791d96b7e9d9efe4cbafae2311c88/pkg/kubelet/prober/worker.go#L268-L271). Hence, the pod will effectively always require more than 30 seconds to become ready. Which is also confirmed by local testing.

> The other 2 are the same as the default setting, but it doesn't seem like a bad thing to have those values configurable with the rest.

Sure, makes totally sense to me.

Reply:
can the kvstoremesh ever transition from Ready to NotReady? If not, a StartupProbe may be the correct answer.

Reply:
> can the kvstoremesh ever transition from Ready to NotReady? 

Currently not (same for the clustermesh-apiserver). And likely not even in the near future, while we may have a readiness probe for the etcd possibly transitioning into NotReady state.

>  If not, a StartupProbe may be the correct answer.

I agree, but the main issue with the startup probe is that it behaves like a liveness probe, restarting the container in case of failure, which would cause a lot of issues in this case -- as the synchronization would need to be performed again from scratch. The only possibility I see could be setting an extremely high failureThreshold, as depending on the number of QPS and cluster size the initial synchronization may take in the order of several (tens of) minutes -- although the bootstrap QPS logic implemented here helps a lot -- which is suboptimal either.

Reply:
Given that info, I've updated the chart to use all default values for the probe unless provided via helm values